core/shared/src/test/scala/fs2/StreamSuite.scala

```diff
@@ -1024,6 +1024,18 @@ class StreamSuite extends Fs2Suite {
     assert(compileErrors("Stream.eval(IO(1)).through(p)").nonEmpty)
   }
 
+  test("Unintended unchunking when using StreamLowPriority#monadInstance") {
+    def generateTuple(pipe: Pipe[fs2.Pure, Int, (Int, Int)]) =
+      (Stream(1, 2, 3) ++ Stream(4, 5, 6)).through(pipe).chunks.toList
+    assertEquals(generateTuple(_.map(e => e -> (e + 1))), generateTuple(_.fproduct(_ + 1)))
+  }
+
+  test("Unintended unchunking when using Stream#monadErrorInstance") {
+    def generateTuple(pipe: Pipe[fs2.Pure, Int, (Int, Int)]) =
+      (Stream(1, 2, 3) ++ Stream(4, 5, 6)).covary[Pure].through(pipe).chunks.compile.toList
+    assertEquals(generateTuple(_.map(e => e -> (e + 1))), generateTuple(_.fproduct(_ + 1)))
+  }
```

Comment:
sorry if I missed it, is this using `MonadError` instance? Since it also appears to be a `Pure` stream :)

Reply:
Sorry, I don't know it will not use the `MonadError` instance before.😂
I changed it to `IO` and locally printed some logs, and found it will use the `MonadError`.
As for the assertEquals, I found that two 'IO' values can't be compared. So I use unsafeRunSync(). Is it right?

Reply:
No problem! :) Instead of using `unsafeRunSync()`, you can do something like:

```scala 
val lhs = generateTuple(_.map(e => e -> (e + 1)))
val rhs = generateTuple(_.fproduct(_ + 1))
(lhs, rhs).mapN(assertEquals(_, _))
```

Reply:
Yeah! This looks more functional.
I changed to that. Thanks!